std_execution.bs

```diff
@@ -4130,16 +4130,14 @@ from the operation before the operation completes.
 
         * <i>count</i> is greater than `0`, in which case `pop_front()` removes an item from the front of the queue, decrements <i>count</i> by `1`, and returns the item.
 
-    2. <i>Synchronization:</i> This function is thread-safe with respect to `push_back()` and `finish()`. Removing a queue item and decrementing <i>count</i> are executed atomically.
-
     <pre highlight="c++">
     void run_loop::push_back(<i>run-loop-opstate-base</i>* item);
     </pre>
 
     1. <i>Effects:</i> Adds `item` to the back of the queue and increments <i>count</i> by `1`.
 
-    2. <i>Synchronization:</i> This function is thread-safe with respect to `pop_front()` and `finish()`. Adding a queue item and incrementing the <i>count</i> are executed atomically.
-
+    2. <i>Synchronization:</i> This operation synchronizes with the `pop_front` operation that obtains `item`.
+    
```

Comment:
I confess I don't understand how this guarantees that multiple threads can call `push_back` concurrently, or that `push_back` can run concurrently with a call to `pop_front` that returns an object other than `item`.

Reply:
Ah, I forgot an edit.

Reply:
Does this wording imply that the two operations (add item, increment count) are executed as a single atomic operation?

Reply:
Addressed by moving to barrier style wording (https://eel.is/c++draft/thread.barrier#class-4), as per Olivier's suggestion